storage/s3.go

```diff
@@ -95,8 +95,25 @@ func newS3Storage(opts Options, sessProvider func() *session.Session) (*S3, erro
 
 	awsSession := sessProvider()
 
+	// for copy operation aws-sdk-go expects a valid
+	// deserializable response and thus tries to unmarshal request.HTTPResponse.Body
+	// into an xml and this stage causes
+	// 'SerializationError: empty response payload status code: 200'
+	// if the response body is empty.
+	// For our tests, we need the following handler.
+	s3api := s3.New(awsSession)
+	s3api.Handlers.Unmarshal.PushBack(func(r *request.Request) {
```

Comment:
I know that this is required for tests but not 100% sold if this is the only way, honestly.

Reply:
thanks. yes, I also think there should be a better way, let me explore other possibilities.

Reply:
The only clean way of solving this issue seems to make changes to gofakes3 that we use for tests, so it can return same response message as aws s3 for copy operations. I can fork and work on [igungor/gofakes3](https://github.com/igungor/gofakes3). Shall we proceed as such? @igungor

Reply:
That'd be great!

Reply:
@fbarotov is your latest commit final? should i review?

Reply:
@igungor yes it is final, but there are some manual changes to `vendor/gofakes3`. Probably this fix should come after a new tag release from `igungor/gofakes3` following [this](https://github.com/igungor/gofakes3/pull/1) pull request.

Reply:
I somehow missed that pull request. It is merged now and released with tag `v0.0.5`. Could you update go.mod and vendor accordingly?

Reply:
done :)